pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -222,6 +223,11 @@ class ModelInterpreter implements Serializable {
 
     }
 
+    @NonCPS
+    private String getFlowNodeId() {
+        return script.getContext(FlowNode.class).getId()
```

Comment:
@dwnusbaum this returns the flownode for the org.jenkinsci.plugins.workflow.steps.GetContextStep not the current flow node. 

The unit test passes which tripped me up when opening this as it just counts tags not if they are the exact right ones.

Reply:
ah I see you meant scanning is still needed:
https://issues.jenkins.io/browse/JENKINS-72841?focusedId=445970&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-445970

Reply:
Yes sorry I should have mentioned that the node will not be exactly the one you want. I think though the immediate parent of this node will be the one you want, or maybe the immediate grandparent? I would some Pipeline with matrix to see. You might not need to scan.

Reply:
I iterated enclosing blocks and it worked fine.